src/ui/pages/products/addNewProduct.page.ts

```diff
@@ -19,6 +23,137 @@ export class AddNewProductPage extends PortalPage {
+    product.price !== undefined && (await this.setValue(this['Price input'], product.price));
+    product.amount !== undefined && (await this.setValue(this['Amount input'], product.amount));
+    product.notes && (await this.setValue(this['Notes textarea'], product.notes + '<A>'));
+  }
+
+  async fillInputs260Notes(product: Partial<IProduct>) {
+    product.name && (await this.setValue(this['Name input'], product.name));
+    product.manufacturer && (await this.selectDropdownValue(this['Manufacturer dropdown'], product.manufacturer));
+    product.price !== undefined && (await this.setValue(this['Price input'], product.price));
+    product.amount !== undefined && (await this.setValue(this['Amount input'], product.amount));
+    product.notes && (await this.setValue(this['Notes textarea'], product.notes.repeat(26)));
+  }
+
+  async fillInputsLetterAmount(product: Partial<IProduct>) {
+    product.name && (await this.setValue(this['Name input'], product.name));
+    product.manufacturer && (await this.selectDropdownValue(this['Manufacturer dropdown'], product.manufacturer));
+    product.price !== undefined && (await this.setValue(this['Price input'], product.price));
+    product.amount !== undefined && (await this.setValue(this['Amount input'], product.amount + 'l'));
+    product.notes && (await this.setValue(this['Notes textarea'], product.notes));
+  }
+
+  async fillInputsBigAmount(product: Partial<IProduct>) {
+    product.name && (await this.setValue(this['Name input'], product.name));
+    product.manufacturer && (await this.selectDropdownValue(this['Manufacturer dropdown'], product.manufacturer));
+    product.price !== undefined && (await this.setValue(this['Price input'], product.price));
+    await this.setValue(this['Amount input'], 10000);
+    product.notes && (await this.setValue(this['Notes textarea'], product.notes));
+  }
+
+  async fillInputsDecimalAmount(product: Partial<IProduct>) {
+    product.name && (await this.setValue(this['Name input'], product.name));
+    product.manufacturer && (await this.selectDropdownValue(this['Manufacturer dropdown'], product.manufacturer));
+    product.price !== undefined && (await this.setValue(this['Price input'], product.price));
+    await this.setValue(this['Amount input'], 5 / 2);
+    product.notes && (await this.setValue(this['Notes textarea'], product.notes));
+  }
+
+  async verifyNameErrorMessage() {
+    await $(this['Name Notification']).waitForDisplayed();
+  }
+
+  async verifyPriceErrorMessage() {
+    await $(this['Price Notification']).waitForDisplayed();
+  }
+
+  async verifyAmountErrorMessage() {
+    await $(this['Amount Notification']).waitForDisplayed();
+  }
+
+  async verifyNotesErrorMessage() {
+    await $(this['Notes Notification']).waitForDisplayed();
+  }
+
+  async verifyErrorMessage(field: 'Notes' | 'Name' | 'Price' | 'Amount') {
+    await $(this[`${field} Notification`]).waitForDisplayed();
```

Comment:
Один метод для нескольких полей
Для `expectedText` можно сделать enum / обьект и подкидьівать в тесте
```suggestion
type TProductInputFileds = 'Notes' | 'Name' | 'Price' | 'Amount';
  async verifyErrorMessage(field: TProductInputFileds, actualText: sting, expectedText: string) {
    await this.findElement(this['Input error  message'](field)).waitForDisplayed();
    expect(actualText).toBe(expectedText);
```

Reply:
отличное дополнение к тому, что я написал выше. Совмести два комментария и выйдет отличный подход к валидации ошибок